sqlglot/generator.py

```diff
@@ -1130,7 +1135,13 @@ def expressions(self, expression, key=None, flat=False, indent=True, sep=", "):
         if flat:
             return sep.join(self.sql(e) for e in expressions)
 
-        expressions = self.sep(sep).join(self.sql(e) for e in expressions)
+        sql = [self.sql(e) for e in expressions]
+        if len(sql) > 1 and self.leading_comma:
```

Comment:
self._leading_comma should go first so we don't always do the length check

Reply:
done